packages/lodestar/src/network/reqresp/reqResp.ts

```diff
@@ -55,73 +67,45 @@ export class ReqResp implements IReqResp {
 
   start(): void {
     this.controller = new AbortController();
-    for (const method of Object.values(Method)) {
-      for (const encoding of Object.values(ReqRespEncoding)) {
-        this.libp2p.handle(
-          createRpcProtocol(method, encoding),
-          async ({connection, stream}: {connection: Connection; stream: ILibP2pStream}) => {
-            const peerId = connection.remotePeer;
-
-            // TODO: Do we really need this now that there is only one encoding?
-            // Remember the prefered encoding of this peer
-            if (method === Method.Status) {
-              this.peerMetadata.encoding.set(peerId, encoding);
-            }
-
-            try {
-              await handleRequest(
-                {config: this.config, logger: this.logger, libp2p: this.libp2p},
-                this.onRequest.bind(this),
-                stream,
-                peerId,
-                method,
-                encoding,
-                this.controller.signal,
-                this.respCount++
-              );
-              // TODO: Do success peer scoring here
-            } catch {
-              // TODO: Do error peer scoring here
-              // Must not throw since this is an event handler
-            }
-          }
-        );
-      }
+    for (const [method, version, encoding] of protocolsSupported) {
```

Comment:
it seems we handle v1 and v2 all the times, do we manage to address this spec somewhere? "`The v1 method MAY be unregistered at the fork boundary. In the event of a request on v1 for an Altair specific payload, the responder MUST return the InvalidRequest response code.`"

Reply:
when we start the node after `ALTAIR_FORK_EPOCH`, we don't need to handle v1

Reply:
> it seems we handle v1 and v2 all the times, do we manage to address this spec somewhere? "The v1 method MAY be unregistered at the fork boundary. In the event of a request on v1 for an Altair specific payload, the responder MUST return the InvalidRequest response code."

SSZ decoding will fail because phase0 is assumed, so it will reject with an InvalidRequest error

Reply:
> when we start the node after ALTAIR_FORK_EPOCH, we don't need to handle v1

Can I handle that on an upcoming PR? We have to review our fork transition code throughly and in the case you note we also want the node to stop listening to v1 when it crosses the fork transition